### PR TITLE
Change display of EnergyProblem variable

### LINKED ISSUE #321: Implement show for the EnergyProblem type
### What and Why

We can make the display of an EnergyProblem variable look nicer following https://docs.julialang.org/en/v1/manual/types/#man-custom-pretty-printing

### Possible Drawbacks

_No response_

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by gnawin**
# Pull request details

## Describe the changes made in this pull request

Change the display of EnergyProblem variable, now it shows 
- termination_status
- solved
- objective_value

## List of related issues or pull requests

Closes #321 

## Collaboration confirmation

As a contributor I confirm

- [x] I read and followed the instructions in README.dev.md
- [x] The documentation is up to date with the changes introduced in this Pull Request (or NA)
- [x] Tests are passing
- [x] Lint is passing

### CODE REVIEW

**Review by datejada — COMMENTED** [review]
@gnawin thanks! please double-check the case when the model is infesible

**Comment by gnawin** [comment]
> @gnawin thanks! please double-check the case when the model is infesible

The relevant options of the `EnergyProblem` is now reset at the end of `create_model!`

File: src/structures.jl [hunk 1]
```diff
@@ -160,3 +160,12 @@ mutable struct EnergyProblem
         )
     end
 end
+
+"""
+Customize the display of EnergyProblem in the REPL.
+"""
```
Comment (abelsiqueira) [inline src/structures.jl:166]:
```suggestion
```

File: src/structures.jl [hunk 2]
```diff
@@ -160,3 +160,12 @@ mutable struct EnergyProblem
...
+function Base.show(io::IO, ep::EnergyProblem)
+    println(io, "Terminal status: ", ep.termination_status)
+    println(io, "Solved: ", ep.solved)
+    print(io, "Objective_value: ", ep.objective_value)
```
Comment (abelsiqueira) [inline src/structures.jl:170]:
```suggestion
    println(io, "EnergyProblem:")
    println(io, "  - Model created: ", !isnothing(ep.model))
    println(io, "  - Solved: ", ep.solved)
    println(io, "  - Termination status: ", ep.termination_status)
    println(io, "  - Objective_value: ", ep.objective_value)
```

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Thanks. I made some small suggestions.

**Review by abelsiqueira — APPROVED** [review]

**Pull request merged**